code/game/objects/structures/urban.dm

```diff
@@ -554,6 +554,23 @@
 		var/mob/living/carbon/human/H = over
 		storage_compartment.open(H)
 
+/obj/structure/billboard
+	name = "billboard"
+	desc = "A large neon billboard displaying a series of advertisements."
+	icon = 'icons/obj/structure/urban/konyang_billboards.dmi'
+	icon_state = "billboard-blank"
+	var/list/ads = list()
+	var/image/ad_overlay
+
+/obj/structure/billboard/Initialize()
+	. = ..()
+	ads = icon_states(icon)
+	ads -= icon_state
+	var/ad = pick(ads)
```

Comment:
They should all be proper subtypes, so they can be placed in a map if a specific ad is needed.
That would allow them to get actual names and descs too, instead of just "billboard".

And there should be a /obj/random/billboard or something that spawns a random billboard.
This is how potted plants work currently too, and you can look at `/obj/random/pottedplant` and the subtypes of `/obj/structure/flora/pottedplant`.